### PR TITLE
[Non_Sanity_Testing] Added comments to enable Ha service for AGW offload test cases

### LINKED ISSUE #5930: Some non-sanity Test Cases are getting stuck with GRPC error-code "UNIMPLEMENTED"
### Your Environment

- **Version:** Local Development Setup
- **Affected Component:** Access Gateway
- **Affected Subcomponent:** Magma MME/SPGW service
- **Deployment Environment:** Vagrant (AGW)

### Setup Details
The S1APTester ([https://github.com/facebookexperimental/S1APTester](https://github.com/facebookexperimental/S1APTester)) was used along with Magma EPC in Vagrant based setup to reproduce the issue.

### Describe the Issue
This issue is a subset of issue #5842 (Also refer PR #5840 which adds all these non-sanity test cases as part of defs.mk file)

The following test cases are getting stuck with this error on terminal: **gRPC failed with StatusCode.UNIMPLEMENTED:**

```
s1aptests/test_agw_offload_idle_active_ue.py
s1aptests/test_agw_offload_mixed_idle_active_multiue.py
```

In both the scenarios, the grpc call is getting exception while handling function **offload_agw** under class **HaUtil** in file "s1aptests/s1ap_utils.py". The error is as follows:
```
*************NbUiNbuHdlUeRadioCapMsg: Send UE cap msg SUCCESS*********
PDN_TYPE: 1
IPv4 PDN_ADDRESS:192 168 128 12
*************************  Offloading UE at state ECM-CONNECTED
gRPC failed with StatusCode.UNIMPLEMENTED:
^CMakefile:42: recipe for target 'integ_test' failed
make: *** [integ_test] Interrupt
```

2 things should be resolved here:
a) The exception should not occur while handling GRPC call
b) The test case should fail rather than getting stuck, even if exception occurs

The execution logs can be found here:
https://drive.google.com/drive/folders/1KjIO9YX87EkUcJXdKEMItBDnJ5md2TII?usp=sharing

**To Reproduce**
1) Run the test case individually "make integ_test TESTS=<testcase_name>"

### PR DESCRIPTION
**Pull request by VinashakAnkitAman**
## Title
[Non_Sanity_Testing] Added comments to enable Ha service for AGW offload test cases

## Summary
This PR leads to the github issue #5930 

The following non-sanity AGW offload test cases were failing with exception. The problem was that Ha service was not enabled by default. This PR is to add a comment in the test script for making people aware that Ha service needs to be enabled for this test case to run and also the steps to enable the Ha service.

```
s1aptests/test_agw_offload_idle_active_ue.py
s1aptests/test_agw_offload_mixed_idle_active_multiue.py
```

The terminal log after getting the exception in GRPC call:
```
*************NbUiNbuHdlUeRadioCapMsg: Send UE cap msg SUCCESS*********
PDN_TYPE: 1
IPv4 PDN_ADDRESS:192 168 128 12
*************************  Offloading UE at state ECM-CONNECTED
gRPC failed with StatusCode.UNIMPLEMENTED:
^CMakefile:42: recipe for target 'integ_test' failed
make: *** [integ_test] Interrupt
```

## Test Plan
Verified after running the test cases multiple time.
 

Signed-off-by: VinashakAnkitAman <ankit.aman@radisys.com>

### CODE REVIEW

**Comment by themarwhal** [comment]
Hi @VinashakAnkitAman! I recently added the inline Python lint annotation. Would be great if you could address the python doc lint issues! Thanks. :)

**Comment by VinashakAnkitAman** [comment]
> Hi @VinashakAnkitAman! I recently added the inline Python lint annotation. Would be great if you could address the python doc lint issues! Thanks. :)

Thanks @themarwhal for your offline help in understanding how we are using these formatting styles there. I have fixed most of the lint warnings and errors in the modified files. The remaining lint warnings are as follows:
```
lte/gateway/python/integ_tests/s1aptests/test_agw_offload_idle_active_ue.py:58:9: WPS437 Found protected attribute usage: _s1_util
lte/gateway/python/integ_tests/s1aptests/test_agw_offload_idle_active_ue.py:66:9: WPS437 Found protected attribute usage: _s1_util
lte/gateway/python/integ_tests/s1aptests/test_agw_offload_mixed_idle_active_multiue.py:62:13: WPS437 Found protected attribute usage: _s1_util
lte/gateway/python/integ_tests/s1aptests/test_agw_offload_mixed_idle_active_multiue.py:70:13: WPS437 Found protected attribute usage: _s1_util
lte/gateway/python/integ_tests/s1aptests/test_agw_offload_mixed_idle_active_multiue.py:110:9: WPS440 Found block variables overlap: i
lte/gateway/python/integ_tests/s1aptests/test_agw_offload_mixed_idle_active_multiue.py:134:9: WPS440 Found block variables overlap: i
```

I think we should suppress the last 2 warnings from setup.cfg file as these warnings are because of using same variable name **i** in multiple for loop blocks and it does not seem to show invalid usage practice to me.
Rest all warnings related to "protected attribute usage" can be ignored for now as it has been used in this trend for most of the test cases. This should either be suppressed or can be changed later in all the files together as part of formatting.

CC: @ulaskozat

**Comment by themarwhal** [comment]
> > Hi @VinashakAnkitAman! I recently added the inline Python lint annotation. Would be great if you could address the python doc lint issues! Thanks. :)
> 
> Thanks @themarwhal for your offline help on how we are using these formatting styles there. I have fixed most of the lint warnings and errors in the modified files. The remaining lint warnings are as follows:
> 
> ```
> lte/gateway/python/integ_tests/s1aptests/test_agw_offload_idle_active_ue.py:58:9: WPS437 Found protected attribute usage: _s1_util
> lte/gateway/python/integ_tests/s1aptests/test_agw_offload_idle_active_ue.py:66:9: WPS437 Found protected attribute usage: _s1_util
> lte/gateway/python/integ_tests/s1aptests/test_agw_offload_mixed_idle_active_multiue.py:62:13: WPS437 Found protected attribute usage: _s1_util
> lte/gateway/python/integ_tests/s1aptests/test_agw_offload_mixed_idle_active_multiue.py:70:13: WPS437 Found protected attribute usage: _s1_util
> lte/gateway/python/integ_tests/s1aptests/test_agw_offload_mixed_idle_active_multiue.py:110:9: WPS440 Found block variables overlap: i
> lte/gateway/python/integ_tests/s1aptests/test_agw_offload_mixed_idle_active_multiue.py:134:9: WPS440 Found block variables overlap: i
> ```
> 
> I think we should suppress the last 2 warnings from setup.cfg file as these warnings are because of using same variable name **i** in multiple for loop blocks and it does not seem to show invalid usage practice to me.
> Rest all warnings related to "protected attribute usage" can be ignored for now as it has been used in this trend for most of the test cases. This should either be suppressed or can be changed later in all the files together as part of formatting.
> 
> CC: @ulaskozat

Good point. I'll put up a PR to suppress those.

**Comment by themarwhal** [comment]
Wonderful! Looks like all lint annotations are addressed. :) Thanks!

File: lte/gateway/python/integ_tests/s1aptests/test_agw_offload_idle_active_ue.py [hunk 1]
```diff
@@ -11,35 +11,41 @@
...
+        """Cleanup after test case execution"""
         self._s1ap_wrapper.cleanup()
 
     def test_agw_offload_idle_active_ue(self):
-        """ Basic attach/detach test with a single UE """
+        """Test case to offload 1 UE in both active and idle states
+
+        NOTE: Ha service must be enabled for running this test case. Set the
+        parameter 'use_ha' in configuration file /etc/magma/mme.yml to 'true'
+        on magma-dev VM and restart MME to enable the Ha service
```
Comment (ulaskozat) [inline lte/gateway/python/integ_tests/s1aptests/test_agw_offload_idle_active_ue.py]: Ha -> HA

Comment (VinashakAnkitAman) [reply lte/gateway/python/integ_tests/s1aptests/test_agw_offload_idle_active_ue.py]: Done

**Review by ulaskozat — APPROVED** [review]
LGTM

**Pull request merged**